csharp/src/Google.Protobuf/JsonFormatter.cs

```diff
@@ -356,7 +362,8 @@ private static bool IsDefaultValue(FieldDescriptor descriptor, object value)
         /// </summary>
         /// <param name="writer">The writer to write the value to. Must not be null.</param>
         /// <param name="value">The value to write. May be null.</param>
-        public void WriteValue(TextWriter writer, object value)
+        /// <param name="depth">The current indentation depth. Optional when writing non-indented values.</param>
+        public void WriteValue(TextWriter writer, object value, int depth = 0)
```

Comment:
At the time of writing, I had not realized this method was public, so the public API is changed.
This could be breaking when `WriteValue` is called through reflection.

Let me know if this should be avoided, and a new private overload is needed.

Reply:
Binary compatibility is a great point that I hadn't even thought of. Fixed it!